src/com/firebirdberlin/nightdream/ui/NightDreamUI.java

```diff
@@ -465,6 +533,14 @@ public void onResume() {
         vibrantColorDark = 0;
         this.locked = settings.isUIlocked;
 
+        removeCallbacks(backgroundChange);
+        background_images[background_image_active].clearAnimation();
+        background_images[ (background_image_active + 1) % 2].clearAnimation();
+        if (settings.getBackgroundMode() == Settings.BACKGROUND_SLIDESHOW) {
+            loadBackgroundImageFiles();
+        }
+        handler.postDelayed(backgroundChange, 15000 * settings.backgroundImageDuration);
+
```

Comment:
backgroundChange is currently triggered nevertheless if the feature is needed.
Is this sufficient ? 👇 
```suggestion
        if (settings.getBackgroundMode() == Settings.BACKGROUND_SLIDESHOW) {
            loadBackgroundImageFiles();
            handler.postDelayed(backgroundChange, 15000 * settings.backgroundImageDuration);
        }

```